utils/cluster_shrinker/cluster_shrinker.pony

```diff
@@ -0,0 +1,131 @@
+/*
+
+Copyright 2017 The Wallaroo Authors.
+
+Licensed as a Wallaroo Enterprise file under the Wallaroo Community
+License (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+     https://github.com/wallaroolabs/wallaroo/blob/master/LICENSE
+
+*/
+
+"""
+A tool for sending autoscale shrink messages.
+"""
+use "buffered"
+use "net"
+use "files"
+use "wallaroo_labs/bytes"
+use "wallaroo_labs/messages"
+use "wallaroo_labs/options"
+
+actor Main
+  var _conn: (TCPConnection | None) = None
+
+  new create(env: Env) =>
+    try
+      let options = Options(env.args)
+      var x_host: String = ""
+      var x_service: String = "0"
+      var query = false
+      var count = U64(0)
+      var workers = Array[String]
+
+      options
+        .add("external", "e", StringArgument)
+        .add("query", "q", None)
+        .add("count", "c", I64Argument)
+        .add("workers", "w", StringArgument)
+        .add("help", "h", None)
+
+        for option in options do
+          match option
+          | ("external", let arg: String) =>
+            let x_addr = arg.split(":")
+            x_host = x_addr(0)?
+            x_service = x_addr(1)?
+          | ("query", None) =>
+            query = true
+          | ("count", let arg: I64) =>
+            count = arg.u64()
+          | ("workers", let arg: String) =>
+            workers = arg.split(",")
+          | ("help", None) =>
+            env.out.print(
+              """
+              PARAMETERS:
+              -----------------------------------------------------------------------------------
+              --external/-e [Specifies address to send message to]
+              --query/-q [Queries for workers eligible for removal]
+              --count/-c [Specifies count of workers to remove from cluster]
+              --workers/-w [Specifies workers to remove from cluster (provided as a comma-separated list of names)]
+              -----------------------------------------------------------------------------------
+              """)
+            return
+          end
+        end
+
+      let auth = env.root as AmbientAuth
+      let msg = ExternalMsgEncoder.shrink(query, workers, count)
```

Comment:
could i currently do something weird like

--count=2
--workers=w1,w2

where it will maybe end up taking out 2 "random" workers + workers "w1,w2"?

I guess what I am asking is, should we be able to combine the 3 together or is it a "pick one" scenario? Pick one feels safest and least likely to have weird results.

Reply:
I've changed it so you must pick exactly one.